src/main/java/com/google/sps/servlets/CalendarServlet.java

```diff
@@ -13,28 +24,77 @@
  * Servlet for getting or creating the Calendar page for each group
  */
 @WebServlet("/calendar")
-public class CalendarServlet extends HttpServlet {
-
+public class CalendarServlet extends AbstractEventsServlet {
   /**
-   * Display the Calendar page for each group.
+   * Display the Calendar page for each group. Query string must contain groupId.
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    SoyFileSet sfs = SoyFileSet
-        .builder()
-        .add(new File("../../src/main/java/templates/calendar.soy"))
-        .build();
-    SoyTofu tofu = sfs.compileToTofu();
-
-    String out = tofu.newRenderer("templates.calendar.calendarPage").render();
-    response.getWriter().println(out);
+    String groupId = getParameter(request, GROUP_ID_PROPERTY);
+
+    if (Strings.isNullOrEmpty(groupId)){
+      ServletUtils.printBadRequestError(response, CALENDAR_BAD_REQUEST_MESSAGE);
+      return;
+    }
+
+    String calendarId = "";
+    try {
+      calendarId = ServletUtils.getGroupProperty(groupId, GROUP_CALENDARID_PROPERTY);
+    } catch (Exception entityError) {
+      ServletUtils.printBadRequestError(response, ENTITY_ERROR_MESSAGE);
+      return;
+    }
+
+    String out = SoyRendererUtils.getOutputString(CALENDAR_SOY_FILE, CALENDAR_TEMPLATE_NAMESPACE,
+        ImmutableMap.of("calendarId", calendarId, "timezone", TIMEZONE));
```

Comment:
Reference the string constant "calendarId" in the constants file

Reply:
Same with timezone

Reply:
timezone (the property string) is only ever used in this file, so will refrain from adding it to the constants file unless I end up using it again elsewhere